components/esp_https_ota/CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
 idf_component_register(SRCS "src/esp_https_ota.c"
                     INCLUDE_DIRS "include"
-                    REQUIRES esp_http_client
+                    REQUIRES esp_http_client app_update
                     PRIV_REQUIRES log app_update)
```

Comment:
`app_update` can be removed from the "private" requirements list if it's in the public interface requirements list